examples/langchain-langgraph/app.py

```diff
@@ -0,0 +1,779 @@
+import logging
+import json
+
+from flask import Flask, request, jsonify, send_from_directory
+from pydantic import BaseModel, Field
+from typing_extensions import Literal
+from langchain_core.tools import tool
+from langchain.chat_models import init_chat_model
+from langgraph.graph import StateGraph, START
+from langgraph.graph.message import add_messages
+from langgraph.prebuilt import ToolNode, tools_condition
+from openai import OpenAI
+import os
+from typing import TypedDict, Annotated
+from dotenv import load_dotenv
+import asyncio
+import threading
+import uuid
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+API_KEY=os.getenv("OPENAI_API_KEY", "not applicable")
+INFERENCE_SERVER_OPENAI = os.getenv("LLAMA_STACK_SERVER_OPENAI", "http://localhost:8321/v1/openai/v1")
+INFERENCE_MODEL = os.getenv("INFERENCE_MODEL", "ollama/llama3.2:3b")
+GUARDRAIL_MODEL = os.getenv("GUARDRAIL_MODEL", "llama-guard3:8b")
+MCP_TOOL_MODEL = os.getenv("MCP_TOOL_MODEL", "ollama/llama3.2:3b")
+GIT_TOKEN = os.getenv("GIT_TOKEN", "not applicable")
+GITHUB_URL=os.getenv("GITHUB_URL", "not applicable")
+GITHUB_ID=os.getenv("GITHUB_ID", "not applicable")
+
+
+llm = init_chat_model(
+    INFERENCE_MODEL,
+    model_provider="openai",
+    api_key=API_KEY,
+    base_url=INFERENCE_SERVER_OPENAI,
+    use_responses_api=True
+)
+
+# so the underlying OpenAI SDK client exists off of llm from init_chat_model, via llm.root_client, but it is a
+# private attribute; so we open up a separate openai client for moderations
+openaiClient = OpenAI(api_key=API_KEY, base_url=INFERENCE_SERVER_OPENAI)
+
+class State(TypedDict):
+    input: str
+    email_body: str
+    messages: Annotated[list, add_messages]
+    decision: str
+    namespace: str
+    data: str
+    mcp_output: str
+    github_issue: str
+    submissionID: str
+
+# Global dictionary to store state by submission ID
+submission_states: dict[str, State] = {}
+
+@tool
+def sendEmail(subject: str, body: str):
+    """Send an email from with the given subject and body.  Do not use attachments"""
+    logger.info(f"sendEmail called with subject: '{subject}'")
+    # Parse submissionID from subject (format: "submissionID: <value>")
+    submission_id = subject.split("submissionID:", 1)[1].strip().split()[0]
```

Comment:
It seems a little weird and brittle to have the model prepend `submissionID:` to the submission ID (in the `init_message` prompt) just to then parse it back out of that prompt.  If there was more time, I would consider restructuring this flow so that the submission ID was available in code and the model didn't need to worry about it.  I don't think it is worth blocking this PR on though.